packages/eslint-plugin/src/rules/naming-convention-utils/validator.ts

```diff
@@ -49,6 +53,14 @@ function createValidator(
         return -1;
       }
 
+      // for backward compatibility, method and property have higher precedence than other meta selectors
+      if (isMethodOrPropertySelector(a.selector)) {
+        return -1;
+      }
+      if (isMethodOrPropertySelector(b.selector)) {
+        return 1;
+      }
+
```

Comment:
The ordering can be modified by adding a flag bit to `method` and `property` selectors, as suggested in https://github.com/typescript-eslint/typescript-eslint/issues/2860#issuecomment-741604143.
But I think it's bit tricky, and I implemented a simpler one.
This should work well as long as `method` and `property` are the only exceptional cases.

Perhaps this can be removed in a future version.

Reply:
I'm happy to special case this here. It's the easiest way to handle it. We can always expand on it later if we need to!

----

This is almost correct!
This logic will create an unstable ordering when both selectors are method/property - but both aren't the same selector.

If both are method/property (and not the same selector), then the order will be the order as written.

This is technically "consistent" for the most part as config rarely changes, but if someone changes their config and reorders the config then all of a sudden the rule could produce a different result - which will be super confusing and hard to figure out.